packages/e2e-playwright/tests/ui/card/binLookup/panLength/panLength.focus.kcp.spec.ts

```diff
@@ -1,58 +1,54 @@
-import { mocks } from './mocks';
-import { binLookupUrl, getBinLookupMock, turnOffSDKMocking } from '../../cardMocks';
+import { test, expect } from '../../../../../fixtures/card.fixture';
+import { getStoryUrl } from '../../../../utils/getStoryUrl';
+import { URL_MAP } from '../../../../../fixtures/URL_MAP';
+import { binLookupMock } from '../../../../../mocks/binLookup/binLookup.mock';
+import { kcpMockOptionalDateAndCvcWithPanLengthMock } from '../../../../../mocks/binLookup/binLookup.data';
+import { REGULAR_TEST_CARD } from '../../../../utils/constants';
 
-/**
- * NOTE - we are mocking the response until such time as we have a genuine card that returns the properties we want to test
- */
-
-let currentMock = null;
-
-const getMock = val => {
-    const mock = mocks[val];
-    currentMock = getBinLookupMock(binLookupUrl, mock);
-    return currentMock;
+const componentConfig = {
+    brands: ['mc', 'visa', 'amex', 'korean_local_card'],
+    configuration: { koreanAuthenticationRequired: true },
+    countryCode: 'KR'
 };
 
 test.describe('Test how Card Component handles binLookup returning a panLength property for a card with a KCP fields', () => {
-    test.beforeEach(async () => {
-        // use config from panLength.kcp.clientScripts.js
-        // await t.navigateTo(cardPage);
-        // For individual test suites (that rely on binLookup & perhaps are being run in isolation)
-        // - provide a way to ensure SDK bin mocking is turned off
-        await turnOffSDKMocking();
-    });
+    test('#1 Fill out PAN (binLookup w. panLength) see that focus moves to tax number since expiryDate & cvc are optional', async ({
+        card,
+        page
+    }) => {
+        await binLookupMock(page, kcpMockOptionalDateAndCvcWithPanLengthMock);
+
+        await card.goto(getStoryUrl({ baseUrl: URL_MAP.card, componentConfig }));
 
-    test('#1 Fill out PAN (binLookup w. panLength) see that focus moves to tax number since expiryDate & cvc are optional', async () => {
-        // await t.addRequestHooks(getMock('kcpMock'));
-        //
-        // // Wait for field to appear in DOM
-        // await cardPage.numHolder();
-        //
-        // const firstDigits = REGULAR_TEST_CARD.substring(0, 15);
-        // const lastDigits = REGULAR_TEST_CARD.substring(15, 16);
-        //
-        // await cardPage.cardUtils.fillCardNumber(t, firstDigits);
-        //
-        // await t.wait(INPUT_DELAY);
-        //
-        // await cardPage.cardUtils.fillCardNumber(t, lastDigits);
-        //
-        // // Expect focus to be place on tax number field
-        // await t.expect(cardPage.kcpTaxNumberLabelWithFocus.exists).ok();
+        await card.isComponentVisible();
+
+        await card.typeCardNumber(REGULAR_TEST_CARD);
+
+        // Expect UI change - tax number field has focus
+        await expect(card.cardNumberLabelWithFocus).not.toBeVisible();
+        await expect(card.kcpTaxNumberLabelWithFocus).toBeVisible();
```

Comment:
It would be better to leverage the testing API rather than checking for locators with classnames.  There are different API's to check specific input/UI states (focus, readonly, disabled, etc..) If I am not mistaken, that also ensures consistent a11y

In this case something like `await expect(card.kcpTaxNumberInput).toBeFocused();` ideally would work

In the Card model, we ideally should keep the locator of the Input field, and to test its different tests we leverage the testing API. We should not need to keep locators for `withFocus` for example